### PR TITLE
add reflect method invoke of accessor impl

### PR DESCRIPTION
**Pull request by guojian0614**
添加方法名访问实现-利用反射

### CODE REVIEW

**Comment by sparrowzoo** [comment]
please ref  this class  Generator4SetFieldMethodAccessor impl

File: sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+        final MethodAccessor methodAccessor = generator4MethodAccessor.newMethodAccessor(authorDTO.getClass());
+
+
+        long start;
+
+        start = System.currentTimeMillis();
+        for (int i = 0; i < 1000_0000; i++) {
+            authorDTO.setAvatar("哈哈" + i);
+//            final Object getAvatar = methodAccessor.get(authorDTO, "getAvatar");
```
Comment (bigduu) [inline sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java:26]: Please remove those comments

File: sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+//            long temp = i;
+//            methodAccessor.set(authorDTO, "setCent",temp);
+//            final Object getCent = methodAccessor.get(authorDTO, "getCent");
+        }
+        System.out.println("原生：" + (System.currentTimeMillis() - start));
+
+        start = System.currentTimeMillis();
+        for (int i = 0; i < 1000_0000; i++) {
+            methodAccessor.set(authorDTO, "setAvatar", "哈哈" + i);
+//            final Object getAvatar = methodAccessor.get(authorDTO, "getAvatar");
```
Comment (bigduu) [inline sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java:36]: Please remove those comments

File: sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+//            long temp = i;
+//            methodAccessor.set(authorDTO, "setCent",temp);
+//            final Object getCent = methodAccessor.get(authorDTO, "getCent");
+        }
+        System.out.println("反射：" + (System.currentTimeMillis() - start));
+
+        start = System.currentTimeMillis();
+        for (int i = 0; i < 1000_0000; i++) {
+            authorDTO.setAvatar("哈哈" + i);
+//            final Object getAvatar = methodAccessor.get(authorDTO, "getAvatar");
```
Comment (bigduu) [inline sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java:46]: Please remove those comments

File: sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+//            long temp = i;
+//            methodAccessor.set(authorDTO, "setCent",temp);
+//            final Object getCent = methodAccessor.get(authorDTO, "getCent");
+        }
+        System.out.println("原生2：" + (System.currentTimeMillis() - start));
+
+        start = System.currentTimeMillis();
+        for (int i = 0; i < 1000_0000; i++) {
+            methodAccessor.set(authorDTO, "setAvatar", "哈哈" + i);
+//            final Object getAvatar = methodAccessor.get(authorDTO, "getAvatar");
```
Comment (bigduu) [inline sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java:56]: Please remove those comments

File: sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
+package java.com.sparrow.cg.impl;
+
+import com.sparrow.cg.Generator4MethodAccessor;
+import com.sparrow.cg.MethodAccessor;
+import com.sparrow.cg.impl.ReflectGenerator4MechodAccessorImpl;
+import com.sparrow.protocol.dto.AuthorDTO;
+
+public class TestReflectGenerator4MechodAccessorImpl {
```
Comment (bigduu) [inline sparrow-test/src/test/java/java/com/sparrow/cg/impl/TestReflectGenerator4MechodAccessorImpl.java:8]: Could you please use Junit for the test? Feel free to do that.

**Review by bigduu — CHANGES_REQUESTED** [review]
There are some commented code need to remove.

File: sparrow-loader/src/main/java/com/sparrow/cg/impl/ReflectGenerator4MechodAccessorImpl.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            method.setAccessible(true);
+            final String methodNameLowCase = method.getName();
+            cache.put(methodNameLowCase,method);
+        }
+        return new ReflectMethodAccessor(cache);
+    }
+
+    private static class ReflectMethodAccessor implements MethodAccessor {
+
+        private final Map<String, Method> METHOD_CACHE;
```
Comment (bigduu) [inline sparrow-loader/src/main/java/com/sparrow/cg/impl/ReflectGenerator4MechodAccessorImpl.java:34]: If the field is not static, please use the Camel-Case naming style

Comment (bigduu) [reply sparrow-loader/src/main/java/com/sparrow/cg/impl/ReflectGenerator4MechodAccessorImpl.java:34]:
```suggestion
        private  final Map<String, Method> methodCache;
```

File: sparrow-loader/src/main/java/com/sparrow/cg/impl/ReflectGenerator4MechodAccessorImpl.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            invoke(o, methodName, arg);
+        }
+
+        /**
+         * execute method
+         * @param o
+         * @param methodName
+         * @param args
+         * @return
+         */
```
Comment (bigduu) [inline sparrow-loader/src/main/java/com/sparrow/cg/impl/ReflectGenerator4MechodAccessorImpl.java:56]:
```suggestion
        /**
         * Execute the method with the string type method name.
         * @param target  The target object that needs to invoke the method.
         * @param methodName The method name defined in the target object
         * @param args The parameters defined in the methodName
         * @return The original method returned type
         */
```

**Review by bigduu — APPROVED** [review]

**Pull request closed without merging**